### PR TITLE
Add user guide

### PR DESCRIPTION
**Pull request by chiaradiamarcelo**
Left overs of #14.

### CODE REVIEW

File: doc/dialects/elasticsearchsql.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# ElasticSearchSQL SQL Dialect
```
Comment (jakobbraun) [inline doc/dialects/elasticsearchsql.md:1]: rename / move to user_guide/user_guide.md

File: doc/dialects/elasticsearchsql.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+## Uploading the JDBC Driver to EXAOperation
+
+First download the [ElasticSearch JDBC driver](https://www.elastic.co/downloads/jdbc-client).
+
+1. [Create a bucket in BucketFS](https://docs.exasol.com/administration/on-premise/bucketfs/create_new_bucket_in_bucketfs_service.htm)
+1. Upload the driver to BucketFS
+
+## Installing the Adapter Script
+
+Upload the latest available release of [Virtual Schema JDBC Adapter](https://github.com/exasol/virtual-schemas/releases) to Bucket FS.
```
Comment (jakobbraun) [inline doc/dialects/elasticsearchsql.md:16]: wrong link

**Review by jakobbraun — CHANGES_REQUESTED** [review]

File: doc/dialects/elasticsearch_sql_user_guide.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
+# ElasticSearchSQL SQL Dialect
```
Comment (chiaradiamarcelo) [inline doc/dialects/elasticsearch_sql_user_guide.md:1]:
```suggestion
# ElasticSearchSQL Dialect User Guide
```

File: doc/dialects/elasticsearch_sql_user_guide.md [hunk 4]
```diff
@@ -13,7 +13,7 @@ First download the [ElasticSearch JDBC driver](https://www.elastic.co/downloads/
 
 ## Installing the Adapter Script
 
-Upload the latest available release of [Virtual Schema JDBC Adapter](https://github.com/exasol/virtual-schemas/releases) to Bucket FS.
+Upload the latest available release of [ElasticSearch Virtual Schema](https://github.com/exasol/elasticsearch-virtual-schema/releases) to Bucket FS.
```
Comment (jakobbraun) [inline doc/dialects/elasticsearch_sql_user_guide.md:16]:
```suggestion
Upload the latest available release of [ElasticSearch Virtual Schema adapter](https://github.com/exasol/elasticsearch-virtual-schema/releases) to Bucket FS.
```

File: doc/dialects/elasticsearch_sql_user_guide.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+	"book_title": "Effective Java",
+	"pages_count": 377,
+	"author":{
+				"first_name": "Joshua",
+				"last_name": "Bloch"
+	}
+}
+```
+
+You can see the `Author` field consists of the `first_name` and `last_name` fields. This are called "nested fields".
```
Comment (chiaradiamarcelo) [inline doc/dialects/elasticsearch_sql_user_guide.md:81]:
```suggestion
You can see the `Author` field consists of the `first_name` and `last_name` fields. These are called "nested fields".
```

File: README.md [hunk 6]
```diff
@@ -30,8 +30,8 @@ If you want to set up a Virtual Schema for a different database system, please h
 
 ### Information for Users
 
-* [ElasticSearchSQL dialect](doc/dialects/elasticsearchsql.md)
-* [User Guide][user-guide]
+* [ElasticSearchSQL Dialect User Guide](doc/dialects/elasticsearch_sql_user_guide.md)
+* [Virtual Schemas User Guide][user-guide]
```
Comment (jakobbraun) [inline README.md:34]:
```suggestion
* [Virtual Schemas User Guide][user-guide]
* [ElasticSearchSQL Dialect User Guide](doc/dialects/elasticsearch_sql_user_guide.md)
```

**Review by jakobbraun — APPROVED** [review]
approved with sugges

File: doc/dialects/elasticsearch_sql_user_guide.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+For referencing nested fields in the ElasticSearchSQL dialect, you use dot(.) as identifier separator, so if you want to refer to the first name of the author, you would do it as `author.first_name`.
+
+In Exasol however the dot(.) can not be used inside quoted identifiers, which means that `author.first_name` is not allowed. However, as explained above, you should always use quoted identifiers given the case sensitive nature of ElasticSearch.
```
Comment (chiaradiamarcelo) [inline doc/dialects/elasticsearch_sql_user_guide.md:85]:
```suggestion
In Exasol however the dot(.) can not be used inside quoted identifiers, which means that `author.first_name` is not allowed. On the other hand, as explained above, you should always use quoted identifiers given the case sensitive nature of ElasticSearch.
```

File: doc/dialects/elasticsearch_sql_user_guide.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+
+### Nested Identifier conversion
+
+ElasticSearch is a document based database. Among other things, it means you can upload documents with a nested structure fields, as any typical JSON file. For example, consider an ElasticSearch index named `book`, containing the following Joshua Bloch's Effective Java book brief description:
+
+```json
+{
+	"book_title": "Effective Java",
+	"pages_count": 377,
+	"author":{
```
Comment (chiaradiamarcelo) [inline doc/dialects/elasticsearch_sql_user_guide.md:74]:
```suggestion
	"author": {
```

File: doc/dialects/elasticsearch_sql_user_guide.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+				"first_name": "Joshua",
```
Comment (chiaradiamarcelo) [inline doc/dialects/elasticsearch_sql_user_guide.md:75]:
```suggestion
			"first_name": "Joshua",
```

File: doc/dialects/elasticsearch_sql_user_guide.md [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+				"last_name": "Bloch"
```
Comment (chiaradiamarcelo) [inline doc/dialects/elasticsearch_sql_user_guide.md:76]:
```suggestion
			"last_name": "Bloch"
```

**Pull request merged**